pkg/cloudprovider/aws/apis/v1alpha1/provider_validation.go

```diff
@@ -163,3 +174,29 @@ func (a *AWS) validateStringEnum(value, field string, validValues []string) *api
 	}
 	return apis.ErrInvalidValue(fmt.Sprintf("%s not in %v", value, strings.Join(validValues, ", ")), field)
 }
+
+func (a *AWS) validateBlockDeviceMappings() (errs *apis.FieldError) {
+	if a.BlockDeviceMappings == nil {
+		return nil
+	}
+	for i, blockDeviceMapping := range a.BlockDeviceMappings {
+		if blockDeviceMapping.DeviceName == nil {
+			errs = errs.Also(apis.ErrMissingField(fmt.Sprintf("%s[%d].deviceName", blockDeviceMappingsPath, i)))
```

Comment:
I think there's a missing field index helper you can use instead of crafting it yourself.